### PR TITLE
[E-ivan < T422] Implement collections.containsAll

### PR DESCRIPTION
**Pull request by imilinovic**
### Description

Please briefly explain the changes you made here.

### Pull request type

- [ ] Bugfix
- [X] Algorithm/Module
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

######################################

#### Module/Algorithm
- [ ] Core algorithm/module implementation
- [ ] [Query module](https://memgraph.com/docs/memgraph/reference-guide/query-modules) implementation
- [ ] Unit tests
- [ ] End-to-end tests
- [ ] Code documentation
- [ ] README short description
- [ ] Documentation on [memgraph/docs](https://github.com/memgraph/docs)
- [ ] Update GQLALchemy signatures in [query builder](https://github.com/memgraph/gqlalchemy/blob/main/gqlalchemy/graph_algorithms/query_builder.py  ) using [query module signature generator](https://github.com/memgraph/gqlalchemy/blob/main/scripts/query_module_signature_generator.py)

######################################

### CODE REVIEW

File: cpp/collections_module/algorithm/algorithm.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#include <unordered_set>
+#include "mgp.hpp"
+
+void Collections::ContainsAll(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+
+  try {
+    const auto &list1{arguments[0].ValueList()};
```
Comment (antoniofilipovic) [inline cpp/collections_module/algorithm/algorithm.cpp:12]: You don't need reference here as you are creating value list1. References are  like aliases.

File: cpp/collections_module/algorithm/algorithm.cpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    std::unordered_set<mgp::Value> set{list1.begin(), list1.end()};
+    const auto &list2{arguments[1].ValueList()};
```
Comment (antoniofilipovic) [inline cpp/collections_module/algorithm/algorithm.cpp:14]: also you don't need reference here

File: cpp/collections_module/algorithm/algorithm.cpp [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    std::unordered_set<mgp::Value> values{list2.begin(), list2.end()};
+
+    bool contained{true};
+    for (const auto &value : values)
+      if (set.find(value) == set.end()) {
+        contained = false;
+        break;
+      }
```
Comment (antoniofilipovic) [inline cpp/collections_module/algorithm/algorithm.cpp:22]: can you use here one of the following: https://en.cppreference.com/w/cpp/container/unordered_set/operator_cmp

or https://en.cppreference.com/w/cpp/algorithm/all_any_none_of, you check if every element is in another set?

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
Few small changes but good work

File: cpp/collections_module/algorithm/algorithm.cpp (hunk 3 above)
Comment (imilinovic) [reply cpp/collections_module/algorithm/algorithm.cpp]: Changed to all_of

**Pull request merged**